Content.Goobstation.Server/Xenobiology/SlimeGrinder/SlimeGrinderSystem.cs

```diff
@@ -0,0 +1,184 @@
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<ActiveSlimeGrinderComponent, ComponentInit>(OnActiveInit);
+        SubscribeLocalEvent<ActiveSlimeGrinderComponent, ComponentShutdown>(OnActiveShutdown);
+        SubscribeLocalEvent<ActiveSlimeGrinderComponent, UnanchorAttemptEvent>(OnUnanchorAttempt);
+
+        SubscribeLocalEvent<SlimeGrinderComponent, MapInitEvent>(OnInit);
+        SubscribeLocalEvent<SlimeGrinderComponent, ComponentShutdown>(OnShutdown);
+        SubscribeLocalEvent<SlimeGrinderComponent, BeforeUnanchoredEvent>(OnUnanchored);
+        SubscribeLocalEvent<SlimeGrinderComponent, AfterInteractUsingEvent>(OnAfterInteractUsing);
+        SubscribeLocalEvent<SlimeGrinderComponent, ClimbedOnEvent>(OnClimbedOn);
+        SubscribeLocalEvent<SlimeGrinderComponent, PowerChangedEvent>(OnPowerChanged);
+        SubscribeLocalEvent<SlimeGrinderComponent, ReclaimerDoAfterEvent>(OnDoAfter);
+    }
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+
+        var query = EntityQueryEnumerator<ActiveSlimeGrinderComponent, SlimeGrinderComponent>();
+        while (query.MoveNext(out var uid, out _, out var grinder))
+        {
+            grinder.ProcessingTimer -= frameTime;
+
+            if (grinder.ProcessingTimer > 0
+                || !TryComp<SlimeComponent>(grinder.EntityGrinded, out var slime))
+                continue;
+
+            var extractProto = _xenobio.GetProducedExtract((grinder.EntityGrinded.Value, slime));
+            SpawnNextToOrDrop(extractProto, uid);
+
+            QueueDel(grinder.EntityGrinded);
+            grinder.EntityGrinded = null;
+
+            RemCompDeferred<ActiveSlimeGrinderComponent>(uid);
+        }
+    }
+
+    #region  Active Grinding
+
+    private void OnActiveInit(Entity<ActiveSlimeGrinderComponent> activeGrinder, ref ComponentInit args)
+    {
+        if (!TryComp<SlimeGrinderComponent>(activeGrinder, out var grinder))
+            return;
+
+        _jitteringSystem.AddJitter(activeGrinder, -10, 100);
+        _sharedAudioSystem.PlayPvs(grinder.GrindSound, activeGrinder);
+        _ambientSoundSystem.SetAmbience(activeGrinder, true);
+    }
+
+    private void OnActiveShutdown(Entity<ActiveSlimeGrinderComponent> activeGrinder, ref ComponentShutdown args)
+    {
+        RemComp<JitteringComponent>(activeGrinder);
+        _ambientSoundSystem.SetAmbience(activeGrinder, false);
+    }
+
+    private void OnUnanchorAttempt(Entity<ActiveSlimeGrinderComponent> activeGrinder, ref UnanchorAttemptEvent args) =>
+        args.Cancel();
+
+    private void OnPowerChanged(EntityUid uid, SlimeGrinderComponent component, ref PowerChangedEvent args)
+    {
+        if (args.Powered)
+        {
+            if (component.ProcessingTimer > 0)
+                EnsureComp<ActiveSlimeGrinderComponent>(uid);
+        }
+        else
+            RemComp<ActiveSlimeGrinderComponent>(uid);
```

Comment:
Keep the brace styles consistent per if statement, rider normally bitches at you for this
```suggestion
        else
        {
            RemComp<ActiveSlimeGrinderComponent>(uid);
        }